core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -884,13 +884,23 @@ class AutoTuner(
       getPropertyValue(lookup).getOrElse(DEF_SHUFFLE_PARTITIONS).toInt
     // TODO: Need to look at other metrics for GPU spills (DEBUG mode), and batch sizes metric
     if (isCalculationEnabled(lookup)) {
-      val totalSpilledMetrics = appInfoProvider.getSpilledMetrics.sum
-      if (totalSpilledMetrics > 0) {
+      val totalSpilledMetricsForShuffleStages =
+        appInfoProvider.getSpillMetricsForShuffleStages.unzip._2.sum
```

Comment:
For performance reason, we don't need to get the sum. "`sum`" iterates on the entire list.
 We only need to check if any of the values is larger than 0. A predicate like `exists` should work better.